lib/modules/manager/npm/extract/index.ts

```diff
@@ -139,6 +144,19 @@ export async function extractPackageFile(
   const yarnrcYmlFileName = getSiblingFileName(fileName, '.yarnrc.yml');
   const yarnZeroInstall = await isZeroInstall(yarnrcYmlFileName);
 
+  let yarnRules: NpmrcRules;
+  const repoYarnrcYml = await readLocalFile(yarnrcYmlFileName, 'utf8');
+  if (is.string(repoYarnrcYml)) {
+    if (is.string(config.npmrc) && !config.npmrcMerge) {
+      logger.debug(
+        { yarnrcYmlFileName },
+        'Repo .yarnrc.yml file is ignored due to config.npmrc with config.npmrcMerge=false'
+      );
+    } else {
+      yarnRules = convertYarnrcYmlToRules(repoYarnrcYml);
+    }
+  }
+
```

Comment:
Would it maybe easier if we just transform the `yarnrc` to `npmrc`? That way we would not need additional logic for mapping the rules to renovate config.

Reply:
@viceice I just noticed that it was like that before and [you didn't like](https://github.com/renovatebot/renovate/pull/16413#pullrequestreview-1027586525) it. Can you explain a bit why that's the case? Doing the transformation once feels much less invasive than what we have right now.

Reply:
We're trying to reduce our "dependency" on `npmrc`. So by transforming to it here we may make it harder to improve our `npmrc` approach